packages/data-store/src/utils/digitalCredential/MappingUtils.ts

```diff
@@ -0,0 +1,112 @@
+import { AddDigitalCredentialArgs } from '../../types/digitalCredential/IAbstractDigitalCredentialStore'
+import { DigitalCredentialEntity } from '../../entities/digitalCredential/DigitalCredentialEntity'
+import {
+  CredentialMapper,
+  decodeSdJwtVc,
+  DocumentFormat,
+  IVerifiableCredential,
+  IVerifiablePresentation,
+  SdJwtDecodedVerifiableCredentialPayload,
+} from '@sphereon/ssi-types'
+import {
+  CredentialDocumentFormat,
+  CredentialType,
+  DigitalCredential,
+  NonPersistedDigitalCredential,
+} from '../../types/digitalCredential/digitalCredential'
+import { computeEntryHash } from '@veramo/utils'
+import { createHash } from 'crypto'
+
+function determineCredentialType(raw: string): CredentialType {
+  if (CredentialMapper.hasProof(raw)) {
+    if (CredentialMapper.isCredential(raw)) {
+      return CredentialType.VC
+    } else if (CredentialMapper.isPresentation(raw)) {
+      return CredentialType.VP
+    }
+  } else {
+    if (CredentialMapper.isCredential(raw)) {
+      return CredentialType.C
+    } else if (CredentialMapper.isPresentation(raw)) {
+      return CredentialType.P
+    }
+  }
+  throw new Error(`Couldn't determine the type of the credential: ${raw}`)
+}
+
+function determineCredentialDocumentFormat(documentFormat: DocumentFormat): CredentialDocumentFormat {
+  switch (documentFormat) {
+    case DocumentFormat.JSONLD:
+      return CredentialDocumentFormat.JSON_LD
+    case DocumentFormat.JWT:
+      return CredentialDocumentFormat.JWT
+    case DocumentFormat.SD_JWT_VC:
+      return CredentialDocumentFormat.SD_JWT
+    default:
+      throw new Error(`Not supported document format: ${documentFormat}`)
+  }
+}
+
+function getExpiresAt(uniformDocument: IVerifiableCredential | IVerifiablePresentation | SdJwtDecodedVerifiableCredentialPayload): Date | undefined {
+  if ('expirationDate' in uniformDocument) {
+    return new Date(uniformDocument.expirationDate)
+  } else if ('exp' in uniformDocument) {
+    return new Date(uniformDocument.exp)
+  }
+  return undefined
+}
+
+function getIssuedAt(uniformDocument: IVerifiableCredential | IVerifiablePresentation | SdJwtDecodedVerifiableCredentialPayload): Date | undefined {
+  if ('issuanceDate' in uniformDocument) {
+    return new Date(uniformDocument.issuanceDate)
+  } else if (uniformDocument.proof && CredentialMapper.getFirstProof(uniformDocument as IVerifiableCredential | IVerifiablePresentation)) {
+    return new Date(CredentialMapper.getFirstProof(uniformDocument as IVerifiableCredential | IVerifiablePresentation)!.created)
+  } else if ('validFrom' in uniformDocument) {
+    return new Date(uniformDocument['validFrom'])
+  } else if ('iat' in uniformDocument) {
```

Comment:
If not mistaken, this should not be `iat` but `nbf`. Please doublecheck that in VCDM 1.1. If that is the case, suggest to add an else if with `nbf` in front of the `iat`

Reply:
yes. in the VCDM 1 they are mentioning `nbf`. added `nbf` right before `iat`